contrib/src/databases.rs

```diff
@@ -0,0 +1,65 @@
+use rocket::Rocket;
+use rocket::fairing::{Fairing, Info, Kind};
+use rocket::config::{Config, Table, Value};
+use rocket::config::Value::String;
+
+pub struct Databases;
+
+#[cfg(feature = "diesel_sqlite")]
+use diesel::SqliteConnection;
+use diesel::r2d2::ConnectionManager;
+
+#[cfg(feature = "pg_pool")]
+use r2d2_postgres::{PostgresConnectionManager, TlsMode};
+
+#[cfg(any(feature = "diesel_sqlite", feature = "pg_pool"))]
+use r2d2::Pool;
+
+// TODO: Assemble methods to instantiate connection pool based on configured options
+impl Databases {
+    pub fn fairing() -> Self {
+        Databases
+    }
+
+    #[cfg(feature = "diesel_sqlite")]
+    pub fn init_sqlite_pool<'a>(&self, credentials: &'a str) -> Pool<ConnectionManager<SqliteConnection>> {
+        let manager = ConnectionManager::<SqliteConnection>::new(credentials);
+        Pool::new(manager).expect("Unable to create sqlite pool")
+    }
+
+    #[cfg(feature = "pg_pool")]
+    pub fn init_rust_pg_pool<'a>(&self, credentials: &'a str) -> Pool<PostgresConnectionManager> {
+        let manager = PostgresConnectionManager::new(credentials, TlsMode::None).unwrap();
+        Pool::new(manager).expect("Unable to create postgres pool")
+    }
+}
+
+impl Fairing for Databases {
+    fn info(&self) -> Info {
+        Info {
+            name: "Database Connection Pool Scaffold",
+            kind: Kind::Attach,
+        }
+    }
+
+    fn on_attach(&self, mut rocket: Rocket) -> Result<Rocket, Rocket> {
+        for database_config in rocket.config().clone().get_slice("databases").unwrap() {
+            match database_config["adapter"] {
+                String(ref adapter) => {
+                    if adapter == "diesel_sqlite" {
+                        rocket = rocket.manage(self.init_sqlite_pool(database_config["url"].as_str().unwrap()));
+                    } else if adapter == "rust-postgres" {
+                        rocket = rocket.manage(self.init_rust_pg_pool(database_config["url"].as_str().unwrap()));
+                    }
+                },
+                _ => return Err(rocket),
+            };
+        }
```

Comment:
This doesn't quite scale, so any suggestions on refining this are welcome. Also, it works in this example, but I can see a situation where an adapter type isn't declared in the `Rocket.toml` that causes a compilation failure here. Not sure if that situation is possible or likely to happen, though.

Reply:
We want to eventually make this extensible such that anyone can plug in a database adapter even if Rocket doesn't support it out of the box. This approach won't allow that since it hardcodes known adapters and their initialization. Furthermore, this will indeed fail to compile unless all features are enabled, which is not when we want.

I think we want to keep a `HashMap` (or something similar) that maps an adapter name to a function (or maybe a trait object) that can initialize the database pool for that adapter. This way, you can start the `HashMap` with the adapter we have built-in support for using `#[cfg]` on each insert line and, when we make it extensible, simple insert to that `HashMap`.

Reply:
That makes a lot of sense. A more extensible and scalable approach than this, for sure!